src/common/Format.h

```diff
@@ -86,7 +96,9 @@ std::string format(const std::string_view fmtstr, const unordered_map<std::strin
   }
 
   // when the loop is done, tokenStart has to be 0 to indicate all tokens are replaced
+  if (!allowMissingMacro) {
   N_ENSURE_EQ(tokenStart, SIZE_T_LIMIT, "some open token not replaced - wrong format string.");
```

Comment:
this should be still a valid check even if `allowMissingMacro` is on. Because this is basically checking if a token is correctly formed with `{}`. I think your test should still work even if you remove this "if condition"